### PR TITLE
New URLs in project.urls

### PR DESCRIPTION
**Pull request by 1ucian0**
With the sunset of qiskit.org, let's move the links in `[project.urls]`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -62,11 +62,11 @@ csp-layout-pass = [
 all = ["qiskit[qasm3-import,visualization,crosstalk-pass,csp-layout-pass]"]
 
 [project.urls]
-Homepage = "https://qiskit.org"
-Documentation = "https://qiskit.org/documentation"
+Homepage = "https://www.ibm.com/quantum/qiskit"
+Documentation = "https://docs.quantum.ibm.com/api/qiskit"
```
Comment (jakelishman) [inline pyproject.toml]: Maybe call this one "API Reference" and make "Documentation" just https://docs.quantum.ibm.com?  That also matches what https://ibm.com/quantum/qiskit calls the documentation landing page for Qiskit.

**Review by jakelishman — COMMENTED** [review]
:(

File: pyproject.toml (hunk 1 above)
Comment (mtreinish) [reply pyproject.toml]:
```suggestion
Documentation = "https://docs.quantum.ibm.com"
"API Reference" = "https://docs.quantum.ibm.com/api/qiskit"
```

**Review by mtreinish — APPROVED** [review]

**Comment by 1ucian0** [comment]
@Mergifyio backport stable/0.46 stable/0.45

**Pull request merged**